extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -51,19 +48,27 @@ const UpgradeNudge = ( { planName, planPathSlug, postId, postType } ) => (
 	</Warning>
 );
 
-export default withSelect( ( select, { plan: planSlug } ) => {
-	const plan = select( 'wordpress-com/plans' ).getPlan( planSlug );
+export default compose( [
+	withSelect( ( select, { plan: planSlug } ) => {
+		const plan = select( 'wordpress-com/plans' ).getPlan( planSlug );
 
-	// WP.com plan objects have a dedicated `path_slug` field, Jetpack plan objects don't
-	// For Jetpack, we thus use the plan slug with the 'jetpack_' prefix removed.
-	const planPathSlug = startsWith( planSlug, 'jetpack_' )
-		? planSlug.substr( 'jetpack_'.length )
-		: get( plan, [ 'path_slug' ] );
+		// WP.com plan objects have a dedicated `path_slug` field, Jetpack plan objects don't
+		// For Jetpack, we thus use the plan slug with the 'jetpack_' prefix removed.
+		const planPathSlug = startsWith( planSlug, 'jetpack_' )
+			? planSlug.substr( 'jetpack_'.length )
+			: get( plan, [ 'path_slug' ] );
 
-	return {
-		planName: get( plan, [ 'product_name_short' ] ),
-		planPathSlug,
-		postId: select( 'core/editor' ).getCurrentPostId(),
-		postType: select( 'core/editor' ).getCurrentPostType(),
-	};
-} )( UpgradeNudge );
+		return {
+			planName: get( plan, [ 'product_name_short' ] ),
+			planPathSlug,
+			postId: select( 'core/editor' ).getCurrentPostId(),
+			postType: select( 'core/editor' ).getCurrentPostType(),
+		};
+	} ),
+	withDispatch( ( dispatch, { planPathSlug, postId, postType } ) => ( {
+		autosaveAndRedirectToUpgrade: async () => {
+			await dispatch( 'core/editor' ).autosave();
+			window.location.href = getUpgradeUrl( { planPathSlug, postId, postType } );
```

Comment:
You'll need `window.top.location.href` to do the `target="_top"` thing.

Reply:
Good spot, thanks! I tested _without_ `.top` on WP.com (in Calypso, with the Gutenberg iframe), and the URL in browser address bar indeed doesn't change there. 

Fixed by 4d0d180fa.